flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOperationUtils.java

```diff
@@ -156,9 +159,43 @@ public static ColumnFamilyDescriptor createColumnFamilyDescriptor(
 		Preconditions.checkState(!Arrays.equals(RocksDB.DEFAULT_COLUMN_FAMILY, nameBytes),
 			"The chosen state name 'default' collides with the name of the default column family!");
 
+		if (writeBufferManagerCapacity != null) {
+			// It'd be great to perform the check earlier, e.g. when creating write buffer manager.
+			// Unfortunately the check needs write buffer size that was just calculated.
+			sanityCheckArenaBlockSize(options.writeBufferSize(), options.arenaBlockSize(), writeBufferManagerCapacity);
+		}
+
 		return new ColumnFamilyDescriptor(nameBytes, options);
 	}
 
+	/**
+	 * Logs a warning of the arena block size is too high causing RocksDB to flush constantly.
+	 * Essentially, the condition here
+	 * <a href="https://github.com/dataArtisans/frocksdb/blob/49bc897d5d768026f1eb816d960c1f2383396ef4/include/rocksdb/write_buffer_manager.h#L47"/>
+	 * will always be true.
+	 *
+	 * @param writeBufferSize the size of write buffer (bytes)
+	 * @param arenaBlockSizeConfigured the manually configured arena block size
+	 * @param writeBufferManagerCapacity the size of the write buffer manager (bytes)
+	 * @return true is sanity check passes, false otherwise
+	 */
+	public static boolean sanityCheckArenaBlockSize(long writeBufferSize, long arenaBlockSizeConfigured, long writeBufferManagerCapacity) throws IllegalStateException {
+		long defaultArenaBlockSize = RocksDBMemoryControllerUtils.calculateRocksDBDefaultArenaBlockSize(writeBufferSize);
+		long arenaBlockSize = arenaBlockSizeConfigured <= 0 ? defaultArenaBlockSize : arenaBlockSizeConfigured;
+		long mutableLimit = RocksDBMemoryControllerUtils.calculateRocksDBMutableLimit(writeBufferManagerCapacity);
+		if (RocksDBMemoryControllerUtils.validateArenaBlockSize(arenaBlockSize, mutableLimit)) {
+			return true;
+		} else {
+			LOG.warn("RocksDBStateBackend performance will be poor because of the current Flink memory configuration! " +
+					"RocksDB will flush memtable constantly, causing high IO and CPU. " +
+					"Typically the easiest fix is to increase task manager managed memory size. " +
+					"If running locally, see the parameter taskmanager.memory.managed.size. " +
+					"Details: arenaBlockSize {} < mutableLimit {} (writeBufferSize {} arenaBlockSizeConfigured {} defaultArenaBlockSize {} writeBufferManagerCapacity {})",
+					arenaBlockSize, mutableLimit, writeBufferSize, arenaBlockSizeConfigured, defaultArenaBlockSize, writeBufferManagerCapacity);
```

Comment:
```suggestion
			LOG.warn("RocksDBStateBackend performance will be poor because of the current Flink memory configuration! " +
					"RocksDB will flush memtable constantly, causing high IO and CPU. " +
					"Typically the easiest fix is to increase task manager managed memory size. " +
					"If running locally, see the parameter taskmanager.memory.managed.size. " +
					"Details: arenaBlockSize {} < mutableLimit {} (writeBufferSize = {}, arenaBlockSizeConfigured = {}," +
					" defaultArenaBlockSize = {}, writeBufferManagerCapacity = {})",
				arenaBlockSize, mutableLimit, writeBufferSize, arenaBlockSizeConfigured,
				defaultArenaBlockSize, writeBufferManagerCapacity);
```
Line wrapping improvement.